sklearn/metrics/classification.py

```diff
@@ -1596,11 +1601,17 @@ def log_loss(y_true, y_pred, eps=1e-15, normalize=True, sample_weight=None):
     The logarithm used is the natural logarithm (base-e).
     """
     lb = LabelBinarizer()
-    T = lb.fit_transform(y_true)
+    lb.fit(labels) if labels is not None else lb.fit(y_true)
+    if labels is None and len(lb.classes_) == 1:
+        raise ValueError('y_true has only one label. Please provide '
+        'the true labels explicitly through the labels argument.')
```

Comment:
We should also add another `ValueError` if `len(lb.classes_) == 1` and `labels is not None` saying that `labels` should have more than one unique label.